content/blog/pulumi-convert-terraform-improvements/index.md

```diff
@@ -0,0 +1,352 @@
+the configuration will need to be moved to stack files in your
+new Pulumi project (eg `Pulumi.dev.yaml`, `Pulumi.prod.yaml`, etc).
+
+You can convert this into a new project with the following command:
+
+{{% chooser language "typescript,python,go" %}}
+
+{{% choosable language typescript %}}
+
+```shell
+pulumi convert --from terraform --language typescript --out ts
+```
+
+I have opted to output it to a different directory to preserve the contents of my Terraform project.
+
+### Project structure
+
+If we navigate to the `ts` directory, we can see a few things:
+
+- index.ts, our equivalent to main.tf for typescript projects
+- sdks directory, where the generated bridged provider is output
+- other node project artifacts such as package.json, node_modules, tsconfig.json, etc.
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```shell
+pulumi convert --from terraform --language python --out python
+```
+
+I have opted to output it to a different directory to preserve the contents of my Terraform project.
+
+### Project structure
+
+If we navigate to the `python` directory, we can see a few things:
+
+- \_\_main\_\_.py, our equivalent to main.tf for python projects
+- sdks directory, where the generated bridged provider is output
+- other python project artifacts (eg requirments.txt, pyproject.toml, etc)
+
+### Python specific cleanup
+
+The python code generator may try to use subscript syntax inappropriately, so you may need to look through the generated code and switch certain things to dot access syntax like so:
+
+```diff
+metadata={
+-        "DB_HOST": db["url"],
++        "DB_HOST": db.url,
+        "DB_USER": "root",
+-        "DB_PASS": db_password["plaintext"],
+-        "DB_NAME": db["name"],
++        "DB_PASS": db_password.plaintext,
++        "DB_NAME": db.name,
+}
+```
+
+{{% /choosable %}}
+
+{{% choosable language go %}}
+
+```shell
+pulumi convert --from terraform --language go --out golang
+```
+
+I have opted to output it to a different directory to preserve the contents of my Terraform project.
+
+### Project structure
+
+If we navigate to the `golang` directory, we can see a few things:
+
+- main.go, our equivalent to main.tf for golang projects
+- sdks directory, where the generated bridged provider is output
+- other golang project artifacts (go.mod, go.sum)
+
+### Go specific cleanup
+
+The go code generator outputs everything in the Terraform code, even if it is
+unused. This is an error in a go program, so I had to manually remove
+unreferenced variables like GCP project, region, and planetscale service token.
+
+These are part of the output because in it's current iteration the code
+converter will convert everything, even if ultimately it is provider
+configuration (see [cleanup](#cleanup)) and not actual code.
+
+{{% /choosable %}}
+
+{{% /chooser %}}
+
+### Cleanup
+
+If we simply run `pulumi preview` this project unfortunately won't run right away.  
```

Comment:
"simply run" -> "run" (grammar: incorrect word use)

Reply:
"project unfortunately won't run right away." -> "project won't run as it is." (simplify grammar)